core/controllers/reader_test.py

```diff
@@ -115,41 +116,46 @@ def setUp(self):
         self.enable_string_classifier = self.swap(
             feconf, 'ENABLE_STRING_CLASSIFIER', True)
 
-        # Reading YAML exploration into a dictionary.
-        yaml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                 '../tests/data/string_classifier_test.yaml')
-        with open(yaml_path, 'r') as yaml_file:
-            self.yaml_content = yaml_file.read()
-
         self.login(self.VIEWER_EMAIL)
         self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
 
         # Load demo exploration.
-        self.exp_id = '0'
+        self.unused_exp_id = '15'
```

Comment:
@pranavsid98 btw, I got confused between string_classifier_test.yaml and classifier_demo_exploration.yaml earlier. I assumed that the data for that PnC exploration was in the demo exploration  - `classifier_demo_exploration.yaml` and so I told you to use exp_id 15 to load it using the load_demo() method but turns out the data is in string_classifier_test.yaml (which is not a demo exploration, sigh!). This makes me think that we should have just one exploration for testing the classifier (to eliminate this confusion) and we should move the data inside `string_classifier_test.yaml` to `classifier_demo_exploration.yaml`. Right now, classifier_demo_exploration.yaml has hardly any data so it doesn't really makes sense.